src/order/rel_iso.lean

```diff
@@ -157,6 +157,22 @@ instance : has_coe (r ↪r s) (r →r s) := ⟨to_rel_hom⟩
 -- see Note [function coercion]
 instance : has_coe_to_fun (r ↪r s) := ⟨λ _, α → β, λ o, o.to_embedding⟩
 
+/--
+To define an relation embedding from an antisymmetric relation `r` to a reflexive relation `s` it
+suffices to give a function together with a proof that it satisfies `s (f a) (f b) ↔ r a b`.
+-/
+def of_map_rel_iff (f : α → β) [is_antisymm α r] [is_refl β s]
+  (hf : ∀ a b, s (f a) (f b) ↔ r a b) : r ↪r s :=
+{ to_fun := f,
+  inj' := λ x y h, antisymm ((hf _ _).1 (h ▸ refl _)) ((hf _ _).1 (h ▸ refl _)),
+  map_rel_iff' := hf }
+
+@[simp]
+lemma of_map_rel_iff_coe (f : α → β) [is_antisymm α r] [is_refl β s]
+  (hf : ∀ a b, s (f a) (f b) ↔ r a b) :
+  ⇑(of_map_rel_iff f hf : r ↪r s) = f :=
+rfl
+
```

Comment:
I think these should be further down the file, we haven't set up the basic definitions yet at this point. I think this belongs next to `of_monotone`.

Reply:
Surely constructors for a structure belong closer?

Reply:
Well, you've not done that for `order_embedding` below, nor did the author of `of_monotone`.

The problem with putting things at the top of the file like this is it makes it harder to spot the boilerplate.
You also risk attributes like `@[simps]` misfiring because the simp-normal form hasn't even been set up yet (eg `to_rel_hom_eq_coe`).